### PR TITLE
Cleanup HCM Plugin Implementation to support multiple HCMs per Listener

### PR DESCRIPTION
**Pull request by sam-heilbron**
# Description

1. Ensure HCM plugins can be run per HttpListener.
2. Ensure Extauth plugins can be run per HttpListener.

# Context

## HttpConnectionManager 

The HttpConnectionManager is a special type of NetworkFilter. We used to have 3 plugins which were responsible for modifying the envoy configuration for an HCM: 
1. HCM
2. Tracing
3. AccessLogging

The problem with the previous HCM plugin that we supported (https://github.com/solo-io/gloo/blob/master/projects/gloo/pkg/plugins/hcm/plugin.go#L43) is that it accepted a listener as a parameter. This meant that the plugin was responsible for:
- Including logic to handle both HttpGateway and TcpGateway
- Modifying all filter chains that were produced, to include the settings that were in a single place

As we move towards a HybridGateway API, we will have Listener with multiple HttpGateway's. This means that a single listener will also have multiple HttpConnectionManager settings. Our plugins needed to be updated to support this new case.

To enable this, we did 2 things:
1. Move the HCM translation out of the plugins into a NetworkFilterTranslator
2. Keep all HCM related settings in plugins

This way, a translator can create an HCM network filter from a single HttpGateway, and it can be run several times (for differnt HttpGateways)

## Extauth Filter
The extauth filter, when processing VirtualHosts, used the Listener to find the HttpListener. This will not work for hybrid gateways. Instead, we supply the HttpListener as a parameter directly. Ideally the listener would be renamed to parentListener, or something to make it clear that it should not be used. However,I  wanted to limit the number of affecting lines of code.

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [ ] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: projects/gloo/pkg/plugins/plugin_interface.go [hunk 1]
```diff
@@ -149,6 +151,12 @@ type TcpFilterChainPlugin interface {
 	CreateTcpFilterChains(params Params, in *v1.TcpListener) ([]*envoy_config_listener_v3.FilterChain, error)
 }
 
+// HttpConnectionManager Plugins
+type HttpConnectionManagerPlugin interface {
```
Comment (sam-heilbron) [inline projects/gloo/pkg/plugins/plugin_interface.go:155]: This plugin was previously defined in a separate file (https://github.com/solo-io/gloo/blob/63b79c491251ac4003433809756a1c79d7f792fe/projects/gloo/pkg/plugins/hcm/hcm_plugin_interface.go#L14) I decided to just move this plugin to our shared list of plugins.

side note: I'm not sure how I feel about this even being a plugin, however we had 2 separate implementations, so I left it as such.

File: projects/gloo/pkg/plugins/extauth/plugin.go [hunk 2]
```diff
@@ -69,7 +69,7 @@ func (p *Plugin) ProcessVirtualHost(
 ) error {
 
 	// Ext_authz filter is not configured on listener, do nothing
-	if !p.isExtAuthzFilterConfigured(params.Listener.GetHttpListener(), params.Snapshot.Upstreams) {
+	if !p.isExtAuthzFilterConfigured(params.HttpListener, params.Snapshot.Upstreams) {
```
Comment (sam-heilbron) [inline projects/gloo/pkg/plugins/extauth/plugin.go:72]: This allows us to pass in an HttpListener directly, instead of requiring each plugin to extract it

**Review by nfuden — COMMENTED** [review]
Overall the pull is fairly tightly scoped which is nice and seems reasonable. 
Did not hit approve purely because bulldozer exists. Otherwise LGTM

File: projects/gloo/pkg/translator/network_filters.go [hunk 3]
```diff
@@ -44,24 +73,27 @@ func (h *httpNetworkFilterTranslator) ComputeNetworkFilters(params plugins.Param
...
 	}
 
-	// add the http connection manager filter after all the InAuth Listener Filters
-	httpConnectionManagerFilter := h.computeHttpConnectionManagerFilter(params)
-	networkFilters = append(networkFilters, plugins.StagedNetworkFilter{
-		NetworkFilter: httpConnectionManagerFilter,
-		Stage:         plugins.AfterStage(plugins.AuthZStage),
-	})
+	hcmNetworkFilters := n.hcmNetworkFilterTranslator.ComputeNetworkFilters(params)
+	if len(hcmNetworkFilters) > 0 {
```
[the remark this answers is not in the document]
Comment (sam-heilbron) [reply projects/gloo/pkg/translator/network_filters.go:89]: I like it!

**Comment by bewebi** [comment]
Latest changes look good to me!

File: .github/workflows/trivy-analysis-scheduled.yaml [hunk 4]
```diff
@@ -52,7 +52,7 @@ jobs:
         env:
           SCAN_DIR: _output/scans
           IMAGE_REPO: quay.io/solo-io
-          MIN_SCANNED_VERSION: 'v1.6.0'
+          MIN_SCANNED_VERSION: 'v1.7.0'
```
Comment (sam-heilbron) [inline .github/workflows/trivy-analysis-scheduled.yaml:55]: CI was failing with: `template: shortcodes/readfile.html:3:13: executing "shortcodes/readfile.html" at <readFile>: error calling readFile: file "static/content/gloo-security-scan.docgen" is too big`

**Review by nfuden — APPROVED** [review]
Nice updates looks good to me

**Comment by sam-heilbron** [comment]
just added WIP label since I know Bernie is mid-review. Will remove after he has reviewed. Thanks for the review @nfuden !

**Comment by sam-heilbron** [comment]
synced with @bewebi and he's ok to merge as is

**Pull request merged**